planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/lane_change.cpp

```diff
@@ -0,0 +1,101 @@
+    goal_planner_utils::find_last_lane_change_completed_lanelet(path, lanelet_map, routing_graph);
+  const auto is_lane_changing_path = lane_change_complete_lane.has_value();
+  if (is_state<Started>() || is_state<Executing>()) {
+    if (is_lane_changing_path) {
+      // NOTE(soblin): is it possible that lane_change_complete_lane changes during execution ?
+      return is_state<Started>() ? Executing(std::get<Started>(state_)) : state_;
+    }
+    std::vector<lanelet::Id> lane_ids;
+    for (const auto & point : path.points) {
+      for (const auto & lane_id : point.lane_ids) {
+        if (std::find(lane_ids.begin(), lane_ids.end(), lane_id) == lane_ids.end()) {
+          lane_ids.push_back(lane_id);
+        }
+      }
+    }
+    if (lane_ids.empty()) {
+      return Aborted{};
+    }
+    const auto current_target_lane_id = is_state<Started>()
+                                          ? std::get<Started>(state_).complete_lane()
+                                          : std::get<Executing>(state_).complete_lane();
+    if (
+      lane_ids.front() == current_target_lane_id &&
+      std::find(lane_ids.begin(), lane_ids.end(), goal_lanelet_id) != lane_ids.end()) {
+      const auto & exec =
+        is_state<Started>() ? Executing{std::get<Started>(state_)} : std::get<Executing>(state_);
+      return Completed{exec};
+    }
+    return Aborted{};
+  }
+
+  // Aborted or NotLaneChanging
+  return is_lane_changing_path ? State{Started{goal_lanelet_id, now}} : State{NotLaneChanging{}};
+}
+
+bool LaneChangeContext::is_not_consistent_transition(const State & from, const State & to)
+{
+  if (is_state<Started>(to) || is_state<Aborted>(to)) {
+    return true;
+  }
+  if (is_state<Executing>(to)) {
+    if (is_state<Aborted>(from) || is_state<NotLaneChanging>(from)) {
+      return true;
+    }
+    if (is_state<Started>(from) || is_state<Executing>(from)) {
+      const auto lane_change_target_to = std::get<Executing>(to).complete_lane();
+      const auto lane_change_target_from = is_state<Started>(from)
+                                             ? std::get<Started>(from).complete_lane()
+                                             : std::get<Executing>(from).complete_lane();
+      if (lane_change_target_to != lane_change_target_from) {
+        return true;
+      }
+      const auto & from_start_time = is_state<Started>(from)
+                                       ? std::get<Started>(from).start_time()
+                                       : std::get<Executing>(from).start_time();
+      const auto & to_start_time = std::get<Executing>(to).start_time();
+      // if timestamp of `to` is larger than `from`, `to` is another lane change after `from`
+      return (to_start_time - from_start_time).seconds() > 0.0;
+    }
+  }
+  if (is_state<NotLaneChanging>(to) && !is_state<NotLaneChanging>(from)) {
+    return true;
+  }
+  // Completed and NotLaneChanging -> NotLaneChanging transition is consitent
```

Comment:
```suggestion
  // Completed and NotLaneChanging -> NotLaneChanging transition is consistent
```
typo

Reply:
```suggestion
  // {Completed} and {NotLaneChanging -> NotLaneChanging} transition are also consistent
```
is easier to understand?

`also` implies other consistent transition case like {Started -> Executing}